### PR TITLE
Fixes Apache LIBCLOUD-809

### PR DESCRIPTION
**Pull request by ghost**
## Changes 
Fixes Apache LIBCLOUD-809

### Description
Merged variables and docstrings for deploy_node to meet features added to create_node.

Reference link: https://issues.apache.org/jira/browse/LIBCLOUD-809

### Status
- done, ready for review

### Checklist (tick everything that applies)

- [X] [Code linting](http://libcloud.readthedocs.org/en/latest/development.html#code-style-guide) (required, can be done after the PR checks)
- [X] Documentation
- [x] [Tests](http://libcloud.readthedocs.org/en/latest/testing.html)
- [ ] [ICLA](http://libcloud.readthedocs.org/en/latest/development.html#contributing-bigger-changes) (required for bigger changes)

### CODE REVIEW

**Comment by ghost** [comment]
+CC @erjohnso 
Would you please be able to take a look?
Thanks!
-alxndr@ (Cloud Graphite)

File: libcloud/compute/drivers/gce.py [hunk 1]
```diff
@@ -6007,9 +6007,18 @@ def ex_set_node_scheduling(self, node, on_host_maintenance=None,
 
         return success
 
-    def deploy_node(self, name, size, image, script, location=None,
-                    ex_network='default', ex_tags=None,
-                    ex_service_accounts=None):
```
Comment (pquentin) [inline libcloud/compute/drivers/gce.py:6012]: I would simply add `, **kwargs` to the list of arguments here.

File: libcloud/compute/drivers/gce.py [hunk 2]
```diff
@@ -6049,6 +6091,62 @@ def deploy_node(self, name, size, image, script, location=None,
...
+        :type     ex_preemptible: ``bool`` or ``None``
+
+        :keyword  ex_image_family: Determine image from an 'Image Family'
+                                   instead of by name. 'image' should be None
+                                   to use this keyword.
+        :type     ex_image_family: ``str`` or ``None``
+
+        :keyword  ex_labels: Labels dictionary for instance.
+        :type     ex_labels: ``dict`` or ``None``
+
```
Comment (pquentin) [inline libcloud/compute/drivers/gce.py:6149]: I would only document `script`, and mention that all other arguments and keyword arguments are documented in `create_node`.

File: libcloud/compute/drivers/gce.py [hunk 3]
```diff
@@ -6058,9 +6156,25 @@ def deploy_node(self, name, size, image, script, location=None,
         metadata = {'items': [{'key': 'startup-script', 'value': script_data}]}
 
         return self.create_node(name, size, image, location=location,
-                                ex_network=ex_network, ex_tags=ex_tags,
-                                ex_metadata=metadata,
-                                ex_service_accounts=ex_service_accounts)
```
Comment (pquentin) [inline libcloud/compute/drivers/gce.py:6063]: I would simply add `, **kwargs` to the arguments.

This means old uses of `deploy_node` will continue to work, and at the same time we will automatically support new `create_node` arguments.

**Review by pquentin — CHANGES_REQUESTED** [review]
Thanks for the change! deploy_node needs to stay in sync with create_node, so we definitely had a problem here.

However, I think we want to stop copy pasting, see the inline comments for details.

What do you think?

**Comment by pquentin** [comment]
@spyrdzy Anything I can do to help get this merged? Do you think my comments make sense? Thanks.

**Comment by ghost** [comment]
Thanks @pquentin !
Sorry for the delay on this, I'll work on getting this ready to pull this week.
Comments definitely mane :)
-Cheers!

**Comment by Kami** [comment]
First thank you for your contribution.

This PR is very old and out of date now so I will go ahead and close it.

If it's still relevant, please sync your changes with the latest master / trunk and open a new PR. Thanks!

**Comment by ghost** [comment]
Thank you and sorry!

**Pull request closed without merging**